src/AbstractShop.java

```diff
@@ -0,0 +1,80 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public abstract class AbstractShop implements IShop {
+    /**
+     * hashMap contains products and it's unique id
+     */
+     protected HashMap<Integer,IProduct> productMap = new HashMap<>();
+    /**
+     * static member to remember unique id
+     */
+    protected static Integer count = 1;

```

Comment:
why is this field?

Reply:
the count is for remembering current free id

Reply:
That's clear,  thanks